src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.cs

```diff
@@ -1361,15 +1434,31 @@ public SolutionCompilationState WithFrozenSourceGeneratedDocuments(
                     // Just compute the checksum from the source text passed in.
                     originalSourceTextChecksum: null,
                     generationDateTime);
-                documentStates.Add(newGeneratedState);
+
+                if (syntaxNode != null)
+                {
+                    newGeneratedState = newGeneratedState.WithSyntaxRoot(syntaxNode);
+                }
+
+                documentStates.Add(newGeneratedState.Id, newGeneratedState);
+            }
+        }
+
+        if (FrozenSourceGeneratedDocumentStates is not null)
```

Comment:
Jason, would tehre be any harm in this always being non-null (but potentially empty)?  It seems like we have to do a lot of null checking that might simplify.

Reply:
At this point no harm; the only real reason it was kept nullable was because it made it clear that things couldn't be frozen twice, but that's now moot with this change.